Kentor.AuthServices.Tests/WebSSO/AcsCommandTests.cs

```diff
@@ -86,7 +86,31 @@ public void AcsCommand_Run_ErrorOnIncorrectXml()
 
             a.ShouldThrow<BadFormatSamlResponseException>()
                 .WithMessage("The SAML response contains incorrect XML")
-                .WithInnerException<XmlException>();
+                .WithInnerException<XmlException>()
+                .Where(ex => ex.Data["Saml2Response"] as string == "<foo />");
+        }
+
+        [TestMethod]
+        public void AcsCommand_Run_ResponseIncludedInException()
+        {
+            string payload =
+                @"<saml2p:Response xmlns:saml2p=""urn:oasis:names:tc:SAML:2.0:protocol""
+                xmlns:saml2=""urn:oasis:names:tc:SAML:2.0:assertion""
+                ID = ""AcsCommand_Run_SuccessfulResult"" Version=""2.0"" />";
```

Comment:
Change the ID to be the name of the test, to avoid problems with replay detection.